test/functional/launch/shared_mem/tests/test-launch-DynamicMem.hpp

```diff
@@ -0,0 +1,110 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_LAUNCH_DYNAMIC_MEM_HPP__
+#define __TEST_LAUNCH_DYNAMIC_MEM_HPP__
+
+#include <numeric>
+
+template <typename INDEX_TYPE, typename WORKING_RES, typename LAUNCH_POLICY, typename TEAM_POLICY, typename THREAD_POLICY>
+void LaunchDynamicMemTestImpl(INDEX_TYPE block_range, INDEX_TYPE thread_range)
+{
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> outer_range(RAJA::stripIndexType(INDEX_TYPE(0)), RAJA::stripIndexType(block_range));
+  RAJA::TypedRangeSegment<INDEX_TYPE> inner_range(RAJA::stripIndexType(INDEX_TYPE(0)), RAJA::stripIndexType(thread_range));
+
+  camp::resources::Resource working_res{WORKING_RES::get_default()};
+  INDEX_TYPE* working_array;
+  INDEX_TYPE* check_array;
+  INDEX_TYPE* test_array;
+  
+  size_t data_len = RAJA::stripIndexType(block_range)*RAJA::stripIndexType(thread_range);
+  
+  allocateForallTestData<INDEX_TYPE>(data_len,
+                                     working_res,
+                                     &working_array,
+                                     &check_array,
+                                     &test_array);
+  
+  
+  for(int b=0; b<RAJA::stripIndexType(block_range); ++b) {
+    for(int c=0; c<RAJA::stripIndexType(thread_range); ++c) {
+      int idx = c + RAJA::stripIndexType(thread_range)*b;
+      test_array[idx] = INDEX_TYPE(b);
+    }
+  }
+
+  size_t shared_mem_size = 1*sizeof(INDEX_TYPE);
+
+  RAJA::launch<LAUNCH_POLICY>
+    (RAJA::LaunchParams(RAJA::Teams(RAJA::stripIndexType(block_range)),
+                        RAJA::Threads(RAJA::stripIndexType(thread_range)), shared_mem_size),
+     [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+      RAJA::loop<TEAM_POLICY>(ctx, outer_range, [&](INDEX_TYPE bid) {
+
+          INDEX_TYPE * tile_ptr = ctx.getSharedMemory<INDEX_TYPE>(1);
+          RAJA::View<INDEX_TYPE, RAJA::Layout<1>> Tile(tile_ptr, 1);
+
+
+          RAJA::loop<THREAD_POLICY>(ctx, RAJA::TypedRangeSegment<INDEX_TYPE>(0,1), [&](INDEX_TYPE ) {
+              Tile(0) = bid;
```

Comment:
For example if you got a buffer of size thread_range,
you could do `Tile(thread_range-tid-1) = thread_range-tid-1 + RAJA::stripIndexType(thread_range)*b` here,
then use `working_array[RAJA::stripIndexType(idx)] = Tile(tid);` below.

Reply:
Sure, that works, just pushed it up.